### PR TITLE
Feature/bookmark

### PR DESCRIPTION
**Pull request by eloigrndl**
Bookmark done and tested

### CODE REVIEW

File: app/src/main/java/com/ncnf/bookmark/ui/FragmentSelector.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ */
+public class FragmentSelector extends FragmentStateAdapter {
+
+    private static final String[] TAB_TITLES = new String[]{"Saved Events", "Organized Events"};
+
+    public FragmentSelector(FragmentActivity activity) {
+        super(activity);
+    }
+
+    public CharSequence getPageTitle(int position) {
```
Comment (HyPn0TiiK) [inline app/src/main/java/com/ncnf/bookmark/ui/FragmentSelector.java:30]: Perform bound check?

**Review by HyPn0TiiK — APPROVED** [review]
Good job mate

File: app/src/main/java/com/ncnf/bookmark/ui/FragmentSelector.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                return new EventDisplayFragment(OWNED_EVENTS_KEY);
+
+            default:
+                return new EventDisplayFragment(SAVED_EVENTS_KEY);
+        }
+    }
+
+    @Override
+    public int getItemCount() {
+        return 2;
```
[the remark this answers is not in the document]
Comment (eloigrndl) [reply app/src/main/java/com/ncnf/bookmark/ui/FragmentSelector.java]: done !

File: app/src/main/java/com/ncnf/bookmark/ui/FragmentSelector.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    }
+
+    public CharSequence getPageTitle(int position) {
+        return TAB_TITLES[position];
+    }
+
+    @NonNull
+    @Override
+    public Fragment createFragment(int position) {
+        switch(position){
```
[the remark this answers is not in the document]
Comment (eloigrndl) [reply app/src/main/java/com/ncnf/bookmark/ui/FragmentSelector.java]: fixed !

File: app/src/main/java/com/ncnf/bookmark/ui/FragmentSelector.java (hunk 1 above)
Comment (eloigrndl) [reply app/src/main/java/com/ncnf/bookmark/ui/FragmentSelector.java:33]: done !

File: app/src/main/java/com/ncnf/bookmark/EventDisplayFragment.java [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        lManager = new LinearLayoutManager(getActivity());
+        recycler.setLayoutManager(lManager);
+
+        // Set the custom adapter
+        adapter = new EventAdapter(eventsToDisplay, this);
+        recycler.setAdapter(adapter);
+        getEventList(view.findViewById(R.id.SavedEventsRecyclerView));
+    }
+
+    private void getEventList(View view){
```
[the remark this answers is not in the document]
Comment (eloigrndl) [reply app/src/main/java/com/ncnf/bookmark/EventDisplayFragment.java:77]: haven't yet thought of it. might change it later for a cleaner code

File: app/src/main/java/com/ncnf/user/PrivateUser.java [hunk 5]
```diff
@@ -25,22 +26,47 @@
...
+
+    public PrivateUser(String UUID, DatabaseService db){
+        this.path = USERS_COLLECTION_KEY + UUID;
+        this.UUID = UUID;
         this.db = db;
-        this.path = path;
+    }
+
+    protected PrivateUser(DatabaseService db, String UUID, String email) {
+        if(UUID == null || email == null || UUID.isEmpty() || email.isEmpty()) {
```
[the remark this answers is not in the document]
Comment (eloigrndl) [reply app/src/main/java/com/ncnf/user/PrivateUser.java]: done !

**Review by Jnf14 — APPROVED** [review]
Good job ! Please just add the DSstore files to the gitignore file.

**Pull request merged**